tomviz/LoadDataReaction.cxx

```diff
@@ -211,6 +211,11 @@ DataSource* LoadDataReaction::loadData(const QStringList& fileNames,
         pxm->NewProxy(node.attribute("xmlgroup").as_string(),
                       node.attribute("xmlname").as_string()));
       if (tomviz::deserialize(reader, node)) {
+        // Use fileNames provided rather than relying on the XML. When loading
+        // a state file the file names provide are resolved relative to the
+        // state files so will survive a directory move.
+        vtkSMPropertyHelper fileNamesHelper(reader, "FileNames");
+        fileNamesHelper.Set(fileName.toLatin1().data());
```

Comment:
@cryos I am overriding the "FileNames" property here, to ensure we get file paths that are resolved relative to the state file location. Its not particularly nice that we have a fileName property in the JSON and one in the pvXmlReader XML, particularly as they are now going to be different. Thoughts?